api/machines.go

```diff
@@ -62,6 +80,54 @@ func (mr *machinesResource) ServeHTTP(rw http.ResponseWriter, req *http.Request)
 	sendResponse(rw, http.StatusOK, page)
 }
 
+func (mr *machinesResource) patch(rw http.ResponseWriter, req *http.Request) {
+	ops := make([]machineMetadataOp, 0)
```

Comment:
I don't know if you actually need to `make` this if you're going to decode JSON into it.